src/data/literal.rs

```diff
@@ -320,4 +326,34 @@ mod tests {
         matches!(Null.concat(Boolean(true)), Null);
         matches!(Null.concat(Null), Null);
     }
+
+    #[test]
+    fn divide() {
+        use crate::data::interval::Interval as I;
+
+        macro_rules! num {
+            ($num: expr) => {
+                Number(Cow::Owned($num.to_owned()))
+            };
+        }
+
+        macro_rules! itv {
+            ($itv: expr) => {
+                Interval(I::Microsecond($itv))
+            };
+        }
+
+        let num_divisor = |x: &str| Number(Cow::Owned(x.to_owned()));
+
+        assert_eq!(num!("12").divide(&num_divisor("2")).unwrap(), num!("6"));
+        assert_eq!(num!("12").divide(&num_divisor("2.0")).unwrap(), num!("6"));
+        assert_eq!(num!("12.0").divide(&num_divisor("2")).unwrap(), num!("6"));
+        assert_eq!(num!("12.0").divide(&num_divisor("2.0")).unwrap(), num!("6"));
+        assert_eq!(itv!(12).divide(&num_divisor("2")).unwrap(), itv!(6));
+        assert_eq!(itv!(12).divide(&num_divisor("2.0")).unwrap(), itv!(6));
+        matches!(num!("12").divide(&Null).unwrap(), Null);
+        matches!(itv!(12).divide(&Null).unwrap(), Null);
+        matches!(Null.divide(&num_divisor("2")).unwrap(), Null);
+        matches!(Null.divide(&Null).unwrap(), Null);
+    }
```

Comment:
New unit tests for `Literal`! Thanks!

Reply:
Shouldn't we use `assert` macro where we use 'matches'?

Reply:
There's actually a tricky issue.
Because.. `Value::Null != Value::Null`.
In SQL `NULL` can be only compared by `IS NULL`.
`Value` does not implement `Eq`, it only implements `PartialEq` because of this `Null` value.

Reply:
I was also curious about this. Thanks for the question @ever0de  and thanks for the answer @panarch 👍